Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManagerInternalInterface.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/function/function_base.h>
+
+class QAction;
+class QWidget;
+
+namespace AzToolsFramework
+{
+    class EditorAction;
+
+    //! ActionManagerInternalInterface
+    //! Internal Interface to query implementation details for actions.
+    class ActionManagerInternalInterface
+    {
+    public:
+        AZ_RTTI(ActionManagerInternalInterface, "{2DCEB7AB-B07A-4085-B5AF-6EEB37439ED6}");
+
+        //! Retrieve a QAction via its identifier.
+        //! @param actionIdentifier The identifier for the action to retrieve.
+        //! @return A raw pointer to the QAction, or nullptr if the action could not be found.
+        virtual QAction* GetAction(const AZStd::string& actionIdentifier) = 0;
+
+        //! Retrieve a QAction via its identifier (const version).
+        //! @param actionIdentifier The identifier for the action to retrieve.
+        //! @return A raw const pointer to the QAction, or nullptr if the action could not be found.
+        virtual const QAction* GetActionConst(const AZStd::string& actionIdentifier) const = 0;
+
+        //! Retrieve an EditorAction via its identifier.
+        //! @param actionIdentifier The identifier for the action to retrieve.
+        //! @return A raw pointer to the EditorAction, or nullptr if the action could not be found.
+        virtual EditorAction* GetEditorAction(const AZStd::string& actionIdentifier) = 0;
```

Comment:
Nit: This API could still be const itself while returning a non-const `EditorAction*`